File: api/models/user.model.js
import mongoose from "mongoose";

const { Schema, model } = mongoose;

const UserSchema = new Schema(
  {
    username: { type: String, required: true, min: 4, unique: true },
    password: { type: String, required: true },
    name: { type: String, required: true },
    avatar: { type: String },
    age: { type: Number },
    address: { type: String },
    phone: { type: String },
    role: { type: Number, default: 1 },
  },
  {
    timestamps: true,
  }
);

const User = model("User", UserSchema);

export default User;

File: lib/model/user_model.dart
class User {
  final String id;
  final String username;
  final String name;
  final int age;
  final String address;
  final String phone;
  final int role;
  final String createdAt;
  final String updatedAt;

  User({
    required this.id,
    required this.username,
    required this.name,
    required this.age,
    required this.address,
    required this.phone,
    required this.role,
    required this.createdAt,
    required this.updatedAt,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      id: json['_id'],
      username: json['username'],
      name: json['name'],
      age: json['age'],
      address: json['address'],
      phone: json['phone'],
      role: json['role'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }
}
